### PR TITLE
Allow submission of all unit types

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Fixes https://github.com/giantswarm/inago/issues/96

### CODE REVIEW

**Comment by JosephSalisbury** [comment]
The trick here is that unit names are validated to only have at most one '@' symbol in, see: https://github.com/giantswarm/inago/blob/master/controller/validator.go#L113

**Comment by JosephSalisbury** [comment]
While I was testing the timer example, I ended up with:

```
core@core-01 ~ $ ./inagoctl submit timer
2016-03-15 16:59:39.133 | INFO     | context.Background: Succeeded to submit group 'timer'.
core@core-01 ~ $ ./inagoctl status timer
Group  Units                  FDState  FCState  SAState   IP            Machine

timer  timer-example.service  loaded   loaded   inactive  172.17.8.101  b944ede1e63146e0bf1dc7ccd01d4adb
timer  timer-example.timer    loaded   loaded   inactive  172.17.8.101  b944ede1e63146e0bf1dc7ccd01d4adb

core@core-01 ~ $ ./inagoctl start -v timer
2016-03-15 16:59:53.973 | DEBUG    | context.Background: cli: starting start
2016-03-15 16:59:53.987 | DEBUG    | context.Background: controller: handling start
2016-03-15 16:59:53.987 | DEBUG    | context.Background: task: creating task
2016-03-15 16:59:53.987 | DEBUG    | context.Background: task: persisting state for task: 11b691fb-169b-4734-bf79-d66bc0495464
2016-03-15 16:59:53.987 | DEBUG    | context.Background: task: created task: 11b691fb-169b-4734-bf79-d66bc0495464
2016-03-15 16:59:53.987 | DEBUG    | context.Background: controller: handling waiting for task
2016-03-15 16:59:53.988 | DEBUG    | context.Background: task: waiting for final status for task: 11b691fb-169b-4734-bf79-d66bc0495464
2016-03-15 16:59:54.003 | DEBUG    | context.Background: fleet: starting unit 'timer-example.service'
2016-03-15 16:59:54.006 | DEBUG    | context.Background: fleet: starting unit 'timer-example.timer'
2016-03-15 16:59:54.015 | DEBUG    | context.Background: controller: handling waiting for status
2016-03-15 16:59:54.988 | DEBUG    | context.Background: task: fetching state for task: 11b691fb-169b-4734-bf79-d66bc0495464
2016-03-15 16:59:55.988 | DEBUG    | context.Background: task: fetching state for task: 11b691fb-169b-4734-bf79-d66bc0495464
2016-03-15 16:59:56.989 | DEBUG    | context.Background: task: fetching state for task: 11b691fb-169b-4734-bf79-d66bc0495464
2016-03-15 16:59:57.990 | DEBUG    | context.Background: task: fetching state for task: 11b691fb-169b-4734-bf79-d66bc0495464
```

which went on for a while.
I have a gut feeling this is related to https://github.com/giantswarm/inago/issues/91

@zeisss thoughts?

**Comment by zeisss** [comment]
very likely. feel free to dig into this. I haven't had time to check it out yet.

File: example/timer/timer-example.service [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[Unit]
+Description=Inago Timer Example Service
+
+[Service]
+ExecStart=/bin/bash -c "echo Beep"
```
Comment (zeisss) [inline example/timer/timer-example.service:5]: we may want to add a `RemainAfterExit=yes` to this file, so it stays with `active/exited` in systemd.

File: controller/request.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ func NewRequest(config RequestConfig) Request {
 	return req
 }
 
-var unitExp = regexp.MustCompile("@.service")
+var unitExp = regexp.MustCompile("@.")
```
Comment (zeisss) [inline controller/request.go:54]: does that work for `@123`?

Comment (zeisss) [inline controller/request.go:54]: ok, there are tests ^^

**Comment by zeisss** [comment]
LGTM, although the regexp feels weird. but there are tests :confused:

**Pull request merged**